code/datums/diseases/advance/advance.dm

```diff
@@ -424,4 +424,16 @@ var/list/advance_cures = 	list(
 		src << "<a href='?_src_=vars;Vars=\ref[D]'>[D.name] - [D.holder]</a>"
 */
 
+/datum/disease/advance/proc/totalStageSpeed()
+	var/total_stage_speed = 0
+	for(var/i in symptoms)
+		var/datum/symptom/S = i
+		total_stage_speed += S.stage_speed
+	return total_stage_speed
+
+/datum/symptom/heal/proc/Heal(mob/living/M, datum/disease/advance/A)
+	var/get_damage = sqrt(20+A.totalStageSpeed())*(2+rand())
+	M.adjustToxLoss(-get_damage)
+	return 1

```

Comment:
I don't see how moving a proc has anything to do with undefined vars and error, when you are not using any defines.